repro.in

```diff
@@ -223,9 +229,12 @@ function init_chroot(){
 
     # Prepare root chroot
     if [ ! -d "$BUILDDIRECTORY"/root ]; then
+        # Acquire lock early
+        mkdir -p "$BUILDDIRECTORY/root"
+        elock 9 "$BUILDDIRECTORY"/root.lock
+
```

Comment:
Actually with the comment over IRC - updating should be opportunistic - the better option is to:

 - move the locking outside the if/else
 - factor out `pacman -Sy*` from both branches, outside the if/else
 - introduce non-blocking elock and use that _only_ for the pacman update